packages/shared/src/hooks/useOrganizationList.tsx

```diff
@@ -1,39 +1,248 @@
 import type {
+  ClerkPaginatedResponse,
   CreateOrganizationParams,
+  GetUserOrganizationInvitationsParams,
   OrganizationMembershipResource,
   OrganizationResource,
   SetActive,
+  UserOrganizationInvitationResource,
+  UserResource,
 } from '@clerk/types';
+import { useCallback, useMemo, useRef, useState } from 'react';
+import useSWR from 'swr';
+import useSWRInfinite from 'swr/infinite';
 
 import { useClerkInstanceContext, useUserContext } from './contexts';
 
+type UseOrganizationListParams = {
+  userInvitations?:
+    | true
+    | (GetUserOrganizationInvitationsParams & {
+        infinite?: boolean;
+        keepPreviousData?: boolean;
+      });
+};
+
 type OrganizationList = ReturnType<typeof createOrganizationList>;
 
+type CustomSetAction<T = unknown> = (size: T | ((_size: T) => T)) => void;
+type PaginatedDataAPI<T = unknown> = {
+  data: T[];
+  count: number;
+  isLoading: boolean;
+  isFetching: boolean;
+  isError: boolean;
+  page: number;
+  pageCount: number;
+  fetchPage: CustomSetAction<number>;
+  fetchPrevious: () => void;
+  fetchNext: () => void;
+  hasNextPage: boolean;
+  hasPreviousPage: boolean;
+};
+
+// Utility type to convert PaginatedDataAPI to properties as undefined, except booleans set to false
+type PaginatedDataAPIWithDefaults<T> = {
+  [K in keyof PaginatedDataAPI<T>]: PaginatedDataAPI<T>[K] extends boolean ? false : PaginatedDataAPI<T>[K] | undefined;
+};
+
 type UseOrganizationListReturn =
-  | { isLoaded: false; organizationList: undefined; createOrganization: undefined; setActive: undefined }
   | {
-      isLoaded: true;
+      isLoaded: false;
+      organizationList: undefined;
+      createOrganization: undefined;
+      setActive: undefined;
+      userInvitations: PaginatedDataAPIWithDefaults<UserOrganizationInvitationResource>;
+    }
+  | {
+      isLoaded: boolean;
       organizationList: OrganizationList;
       createOrganization: (params: CreateOrganizationParams) => Promise<OrganizationResource>;
       setActive: SetActive;
+      userInvitations: PaginatedDataAPI<UserOrganizationInvitationResource>;
     };
 
-type UseOrganizationList = () => UseOrganizationListReturn;
+type UseOrganizationList = (params?: UseOrganizationListParams) => UseOrganizationListReturn;
+
+export const useOrganizationList: UseOrganizationList = params => {
+  const { userInvitations } = params || {};
+
+  const shouldUseDefaults = typeof userInvitations === 'boolean' && userInvitations;
+  const [paginatedPage, setPaginatedPage] = useState(shouldUseDefaults ? 1 : userInvitations?.initialPage ?? 1);
+
+  // Cache initialPage and initialPageSize until unmount
+  const initialPageRef = useRef(shouldUseDefaults ? 1 : userInvitations?.initialPage ?? 1);
```

Comment:
❓ Do we need the last default fallback? Can't we ensure that the `initialPage` attribute always has a value?

Reply:
Follow-up question? Do we need a ref for this?

Reply:
initialPage does not always have a value as someone can simply pass as params the following 
```ts
useOrganizationList({infinite:true})
```
Here we want to not use the defaults in general but use the default value of initialPage